File: imagepicker/src/main/java/com/esafirm/imagepicker/helper/ConfigUtils.java
package com.esafirm.imagepicker.helper;

import android.content.Context;

import com.esafirm.imagepicker.R;
import com.esafirm.imagepicker.features.ImagePickerConfig;
import com.esafirm.imagepicker.features.IpCons;
import com.esafirm.imagepicker.features.ReturnMode;
import com.esafirm.imagepicker.features.common.BaseConfig;

import java.io.Serializable;

public class ConfigUtils {

    public static ImagePickerConfig checkConfig(ImagePickerConfig config) {
        if (config == null) {
            throw new IllegalStateException("ImagePickerConfig cannot be null");
        }
        if (config.getMode() != IpCons.MODE_SINGLE
                && (config.getReturnMode() == ReturnMode.GALLERY_ONLY
                || config.getReturnMode() == ReturnMode.ALL)) {
            throw new IllegalStateException("ReturnMode.GALLERY_ONLY and ReturnMode.ALL is only applicable in Single Mode!");
        }
        if (config.getImageLoader() != null && !(config.getImageLoader() instanceof Serializable)) {
            throw new IllegalStateException("Custom image loader must be a class that implement ImageLoader." +
                    " This limitation due to Serializeable");
        }
        return config;
    }

    public static boolean shouldReturn(BaseConfig config, boolean isCamera) {
        ReturnMode mode = config.getReturnMode();
        if (isCamera) {
            return mode == ReturnMode.ALL || mode == ReturnMode.CAMERA_ONLY;
        } else {
            return mode == ReturnMode.ALL || mode == ReturnMode.GALLERY_ONLY;
        }
    }

    public static String getFolderTitle(Context context, ImagePickerConfig config) {
        final String folderTitle = config.getFolderTitle();
        return ImagePickerUtils.isStringEmpty(folderTitle)
                ? context.getString(R.string.ef_title_folder)
                : folderTitle;
    }

    public static String getImageTitle(Context context, ImagePickerConfig config) {
        final String configImageTitle = config.getImageTitle();
        return ImagePickerUtils.isStringEmpty(configImageTitle)
                ? context.getString(R.string.ef_title_select_image)
                : configImageTitle;
    }

    public static String getDoneButtonText(Context context, ImagePickerConfig config) {
        final String doneButtonText = config.getDoneButtonText();
        return ImagePickerUtils.isStringEmpty(doneButtonText)
                ? context.getString(R.string.ef_done)
                : doneButtonText;
    }
}

File: imagepicker/src/main/java/com/esafirm/imagepicker/helper/ConfigUtils.kt
package com.esafirm.imagepicker.helper

import android.content.Context
import com.esafirm.imagepicker.R
import com.esafirm.imagepicker.features.ImagePickerConfig
import com.esafirm.imagepicker.features.IpCons
import com.esafirm.imagepicker.features.ReturnMode
import com.esafirm.imagepicker.features.cameraonly.CameraOnlyConfig
import com.esafirm.imagepicker.features.common.BaseConfig

object ConfigUtils {
    fun checkConfig(config: ImagePickerConfig?): ImagePickerConfig {
        checkNotNull(config) { "ImagePickerConfig cannot be null" }
        check(!(config.mode != IpCons.MODE_SINGLE
            && (config.returnMode === ReturnMode.GALLERY_ONLY
            || config.returnMode === ReturnMode.ALL))) { "ReturnMode.GALLERY_ONLY and ReturnMode.ALL is only applicable in Single Mode!" }
        return config
    }

    fun shouldReturn(config: BaseConfig, isCamera: Boolean): Boolean {
        if (config is CameraOnlyConfig) return true
        val mode = config.returnMode
        return if (isCamera) {
            mode === ReturnMode.ALL || mode === ReturnMode.CAMERA_ONLY
        } else {
            mode === ReturnMode.ALL || mode === ReturnMode.GALLERY_ONLY
        }
    }

    fun getFolderTitle(context: Context, config: ImagePickerConfig): String {
        val folderTitle = config.folderTitle
        return if (folderTitle.isNullOrBlank()) context.getString(R.string.ef_title_folder) else folderTitle
    }

    fun getImageTitle(context: Context, config: ImagePickerConfig): String {
        val configImageTitle = config.imageTitle
        return if (configImageTitle.isNullOrBlank()) context.getString(R.string.ef_title_select_image) else configImageTitle
    }

    fun getDoneButtonText(context: Context, config: ImagePickerConfig): String {
        val doneButtonText = config.doneButtonText
        return if (doneButtonText.isNullOrBlank()) context.getString(R.string.ef_done) else doneButtonText
    }
}